### ISSUE TITLE
Shard doesn't switch to read-only mode when shard_ro_error_threshold reached

### ISSUE THREAD

**Issue by anikeev-yadro** [nspcc-dev/neofs-node#1857]
## Expected Behavior
Shard should switch to read-only mode when when shard_ro_error_threshold reached.

## Current Behavior
Shard doesn't switch to read-only mode when shard_ro_error_threshold reached

## Steps to Reproduce (for bugs)
1. Make some WRITE disk errors on the shard:
```
Oct 07 16:11:44 az neofs-node[18920]: 2022-10-07T16:11:44.499Z        debug        shard/put.go:58        can't put object to the write-cache, trying blobstor        {"shard_id": "Jj5YJnPQTXHSoYg6KpUuQi", "err": "no space left in the write cache"}
Oct 07 16:11:44 az neofs-node[18920]: 2022-10-07T16:11:44.953Z        warn        engine/engine.go:44        could not put object to shard        {"shard_id": "Jj5YJnPQTXHSoYg6KpUuQi", "error count": 1, "error": "could not put object to BLOB storage: write /data/neofs/data0/5/g/P/7/ACHnzhRiLovxkgngXQ93GkTKSC6EBhq6x1VZbRWw.HqbCTJ1WRFwSJMuRqn6W1so4LLQs1HQq3Jve7x8FdFpQ: input/output error"}
```
2. shard_ro_error_threshold reached, but shard doesn't switch to read-obly mode according to config:
```
root@az:/etc/neofs/storage# grep shard_ro_error_threshold config.yml
  shard_ro_error_threshold: 30
```
```
root@az:/etc/neofs/storage# sudo neofs-cli --endpoint 127.0.0.1:8091 -w /etc/neofs/storage/wallet.json control shards list|grep Jj5YJnPQTXHSoYg6KpUuQi -A 6
Enter password >
Shard Jj5YJnPQTXHSoYg6KpUuQi:
Mode: read-write
Metabase: /srv/neofs/meta0/metabase0.db
Blobstor: /data/neofs/data0
Write-cache: /srv/neofs/meta0/write_cache0
Pilorama: /srv/neofs/meta0/pilorama0.db
Error count: 43
```
**Config:**  [config (3).zip](https://github.com/nspcc-dev/neofs-node/files/9735602/config.3.zip)
**Logs:** [shard_err.zip](https://github.com/nspcc-dev/neofs-node/files/9735586/shard_err.zip)

**Versions:**
```
NeoFS Storage node
Version: v0.32.0-73-g01ddb3f8
GoVersion: go1.18.4
```
## Your Environment
Server setup and configuration:
cloud, 4 VMs, 4 SN, 4 http qw, 4 s3 gw

Operating System and version (uname -a):
linux vedi 5.10.0-16-amd64 https://github.com/nspcc-dev/neofs-node/issues/1 SMP Debian 5.10.127-1 (2022-06-30) x86_64 GNU/Linux

**Comment by fyrchik** [comment]
There are many lines like this
```
Oct 07 16:16:14 az neofs-node[18920]: 2022-10-07T16:16:14.476Z        error        engine/engine.go:56        failed to move shard in degraded mode        {"error count": 36, "error": "write /data/neofs/data0/3/S/o/F/xLJbxkD31ho9arpRyi4jQH4RSAVB4GpU4zQ2Gq6t.D8UNmzkDVbkDATiWZ4Liuc21KPnXZ4UtVVaSmTk249HA: input/output error"}
```
What is the error probability for WRITE errors in your setup?

**Comment by anikeev-yadro** [comment]
@fyrchik config for unreliablefs:
```
root@az:/srv# cat unreliablefs.conf
[errinj_noop]
op_regexp = WRITE
path_regexp = .data0*
probability = 90
```

**Comment by fyrchik** [comment]
Because we flush write-cache before moving to a degraded mode, with 90% write fail probability this situation is kinda expected.
I have thought about moving to read-only mode in case of error, but this way administrator won't know about the errors.

What I propose to do is to move write-cache in read-only mode but still return an error occured during moving it into a degraded.
This way it could be a bit more reliable/expected.

**Comment by fyrchik** [comment]
Fixed by #1957 .
